Inc/DirectXMath.h

```diff
@@ -38,60 +40,80 @@
 #endif
 
 #ifndef XM_DEPRECATED
+#ifdef __GNUC__
+#define XM_DEPRECATED __attribute__ ((deprecated))
+#else
 #define XM_DEPRECATED __declspec(deprecated("This is deprecated and will be removed in a future version."))
 #endif
+#endif
 
 #if !defined(_XM_AVX2_INTRINSICS_) && defined(__AVX2__) && !defined(_XM_NO_INTRINSICS_)
 #define _XM_AVX2_INTRINSICS_
 #endif
 
-#if !defined(_XM_FMA3_INTRINSICS_) && defined(_XM_AVX2_INTRINSICS_) && !defined(_XM_NO_INTRINSICS_)
+#if !defined(_XM_FMA3_INTRINSICS_) && (defined(_XM_AVX2_INTRINSICS_) && !defined(__GNUC__)) && !defined(_XM_NO_INTRINSICS_)
+#define _XM_FMA3_INTRINSICS_
+#endif
+
+#if !defined(_XM_FMA3_INTRINSICS_) && defined(__FMA__) && !defined(_XM_NO_INTRINSICS_)
 #define _XM_FMA3_INTRINSICS_
 #endif
 
-#if !defined(_XM_F16C_INTRINSICS_) && defined(_XM_AVX2_INTRINSICS_) && !defined(_XM_NO_INTRINSICS_)
+#if !defined(_XM_F16C_INTRINSICS_) && (defined(_XM_AVX2_INTRINSICS_) && !defined(__GNUC__)) && !defined(_XM_NO_INTRINSICS_)
```

Comment:
Why did you add all these !__GNUC__ guards? These shouldn't be enabled without explicit opt-in in any case.

Reply:
Well:
* User set up `-mavx2` compiler flag, it defines `__AVX2__`
* `__AVX2__` defines `_XM_AVX2_INTRINSICS_`
* `_XM_AVX2_INTRINSICS_` defines `_XM_F16C_INTRINSICS_`
* and here we got compilation error, because for F16C instrinsics we should explicitly set up `-mf16c` compile flag.

Reply:
So, that's why I switched to use compiler definitions for GCC, it much safer.

Reply:
So this is "By Design". If you use AVX2, it assumes you can also use F16C and FMA because they are inclusive. The rules follow all the way down.  In fact, with clang setting ``-mavx2`` implies ``-mf16c``.

Reply:
and since ``__GNUC__`` exists for clang/LLVM, this is likely to break it for clang. Just document that for GNUC you need to use ``-mavx2 -mf16c -mfma`` is safer and avoids breaking clang/LLVM

Reply:
OK, removed